yarn-project/p2p/src/service/ip_query.ts

```diff
@@ -0,0 +1,21 @@
+import * as https from 'https';
+
+export function getPublicIp(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    https
+      .get('https://checkip.amazonaws.com', res => {
+        let data = '';
+
+        res.on('data', chunk => {
+          data += chunk;
+        });
+
+        res.on('end', () => {
+          resolve(data.trim());
+        });
+      })
+      .on('error', err => {
+        reject(`Error fetching public IP: ${err.message}`);
+      });
```

Comment:
What if we used global `fetch` for this?
```suggestion
const response = await fetch('https://checkip.amazonaws.com');
const text = await response.text();
return text.trim();
```